guides/micronaut-data-jdbc-repository/groovy/src/main/groovy/example/micronaut/GenreRepository.groovy

```diff
@@ -0,0 +1,27 @@
+package example.micronaut
+
+import example.micronaut.domain.Genre
+import io.micronaut.core.annotation.NonNull
+import io.micronaut.data.annotation.Id
+import io.micronaut.data.exceptions.DataAccessException
+import io.micronaut.data.jdbc.annotation.JdbcRepository
+import io.micronaut.data.model.query.builder.sql.Dialect
+import io.micronaut.data.repository.PageableRepository
+import javax.transaction.Transactional
+import javax.validation.constraints.NotBlank
+import javax.validation.constraints.NotNull
+
+@JdbcRepository(dialect = Dialect.MYSQL) // <1>
+abstract class GenreRepository implements PageableRepository<Genre, Long> { // <2>
```

Comment:
can yo not use an `interface` with a default method for `saveWithException` as we do in the java guide?

Reply:
Groovy interfaces don't allow default methods. Can use a Trait, will look closer to the Java version; however, have to keep the Kotlin version as an abstract class (see below). Only way to make the all similar is to have them as abstract.